nanoFramework.Json/TimeExtensions.cs

```diff
@@ -126,178 +186,208 @@ public static DateTime FromIso8601(string date)
-		/// </summary>
-		/// <param name="ajax"></param>
-		/// <returns></returns>
-		public static DateTime FromASPNetAjax(string ajax)
-		{
-			string[] parts = ajax.Split(new char[] { '(', ')' });
-
-			long ticks = Convert.ToInt64(parts[1]);
-
-			// Create a Utc DateTime based on the tick count
-			DateTime dt = new(ticks, DateTimeKind.Utc);
-
-			return dt;
-		}
-
-		internal static DateTime ConvertFromString(string value)
+            if (utc)
+            {
+                // Convert the Kind to DateTimeKind.Utc if string Z present
+                dt = new DateTime(dt.Ticks, DateTimeKind.Utc); //TODO!!!
+            }
+            else
+            {
+                //nF does not support non UTC dates, so should we throw an exception instead.
+                throw new NotSupportedException("DateTime is not UTC");
+
+                //TODO: remove old handling of non UTC datetime code.
+                //// If a time offset was specified instead of the UTC marker,
+                //// add/subtract in the hours/minutes
+                //if ((utc == false) && (parts.Length >= 9))
+                //{
+                //	// There better be a timezone offset
+                //	string hourOffset = (parts.Length > 7) ? parts[7] : "";
+                //	string minuteOffset = (parts.Length > 8) ? parts[8] : "";
+                //	if (date.Contains("+"))
+                //	{
+                //		dt = dt.AddHours(Convert.ToDouble(hourOffset));
+                //		dt = dt.AddMinutes(Convert.ToDouble(minuteOffset));
+                //	}
+                //	else
+                //	{
+                //		dt = dt.AddHours(-(Convert.ToDouble(hourOffset)));
+                //		dt = dt.AddMinutes(-(Convert.ToDouble(minuteOffset)));
+                //	}
+                //}
+            }
+
+            return dt;
+        }
+
+        /// <summary>
+        /// Converts a DateTime object into an ISO 8601 string.  This version
+        /// always returns the string in UTC format.
+        /// </summary>
+        /// <param name="dt"></param>
+        /// <returns></returns>
+        public static string ToIso8601(DateTime dt)
+        {
+            string result = dt.Year.ToString() + "-" +
+                            TwoDigits(dt.Month) + "-" +
+                            TwoDigits(dt.Day) + "T" +
+                            TwoDigits(dt.Hour) + ":" +
+                            TwoDigits(dt.Minute) + ":" +
+                            TwoDigits(dt.Second) + "." +
+                            ThreeDigits(dt.Millisecond) + "Z";
+
+            return result;
+        }
+
+        /// <summary>
+        /// Returns true if the given string contains only numeric characters.
+        /// </summary>
+        /// <param name="str"></param>
+        /// <returns></returns>
+        private static bool IsNumeric(string str)
+        {
+            foreach (char c in str)
+            {
+                if (!((c >= '0') && (c <= '9')))
```

Comment:
We probably should have [IsNumber()](https://docs.microsoft.com/en-us/dotnet/api/system.char.isnumber?view=net-5.0) in mscorlib...

But this will work in the meantime.